docs/repo/property-pages/how-to-add-a-new-launch-profile-kind.md

```diff
@@ -103,6 +103,125 @@ And you're done. Projects that import the .targets file will now show this UI wh
 
 ## Option 2: Embedded XAML file
 
-In this option the XAML file is embedded in an assembly as a resource and discovered by means of a MEF export. Compared to Option 1 this requires more initial setup but does not require you to distribute an additional file. This may be an attractive option if you are already exporting MEF components for use in Visual Studio.
+In this option the XAML file is embedded in an assembly as a resource and discovered by means of a MEF export. Compared to Option 1, this requires more initial setup but does not require you to distribute an additional file. This may be an attractive option if you are already exporting MEF components for use in Visual Studio.
 
-_Steps to be determined._
+### Step 1: Define the `XamlPropertyRule` item
+
+Add the following item to your .csproj file:
+
+```xml
+  <ItemGroup>
+    <XamlPropertyRule Include="path\to\MyPropertyPage.xaml">
+      <Namespace>MyNamespace</Namespace>
+      <DataAccess>IRule</DataAccess>
+      <RuleInjectionClassName>MyPropertyPageProperties</RuleInjectionClassName>
+    </XamlPropertyRule>
+  </ItemGroup>
+```
+
+### Step 2: Register the property page and properties
+
+Add a partial class that corresponds to the properties defined within the rule and generated during build:
+
+```csharp
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using System;
+using System.ComponentModel.Composition;
+
+namespace MyNamespace
+{
+    [Export]
+    [AppliesTo(MyProject.UniqueCapability)]
+    internal partial class MyPropertyPageProperties : StronglyTypedPropertyAccess
+    {
+        [ExportPropertyXamlRuleDefinition("<Assembly Name>, Version=<Assembly Version>, Culture=<Assembly Culture>, PublicKeyToken=<Assembly Public Key Token>", "XamlRuleToCode:MyPropertyPage.xaml", "Project")]
+        [AppliesTo(MyProject.UniqueCapability)]
+        private object MyRule { get { throw new NotImplementedException(); } }
+
+        [ImportingConstructor]
+        public MyPropertyPageProperties(ConfiguredProject configuredProject)
+            : base(configuredProject)
+        {
+        }
+
+        public MyPropertyPageProperties(ConfiguredProject configuredProject, string file, string itemType, string itemName)
+            : base(configuredProject, file, itemType, itemName)
+        {
+        }
+
+        public MyPropertyPageProperties(ConfiguredProject configuredProject, IProjectPropertiesContext projectPropertiesContext)
+            : base(configuredProject, projectPropertiesContext)
+        {
+        }
+
+        public MyPropertyPageProperties(ConfiguredProject configuredProject, UnconfiguredProject project)
+            : base(configuredProject, project)
+        {
+        }
+    }
+}
+```
+
+### Step 3: Register the assembly codebase
+
+In order for Visual Studio to locate the rule, you must register the codebase of the assembly embedding the rule from the Visual Studio package via the `ProvideCodeBaseAttribute`.
```

Comment:
Is this true? The `dotnet/project-system` repo exports rules yet doesn't seem to use this attribute.

Reply:
#7462 _removed_ our use of CodeBase. I'll double-check this.

Reply:
The important part is to get the assembly on the VS binding path because we end up loading the assembly to read the resources. I've replaced this with a brief note on how to do this.

Reply:
Is it general practice (now) to use `ProvideBindingPathAttribute` for extensions? We use `ProvideCodeBaseAttribute` in all of our extensions, in part to prevent VS from blindly using assemblies it might find in the folder over the "official" versions shipped with VS core, which has caused occasional problems in the past.